src/decoder.ts

```diff
@@ -293,6 +293,43 @@ export class Decoder<A> {
     });
   }
 
+  static strictObject<A>(decoders?: DecoderObject<A>) {
+    const decoderKeys = isJsonObject(decoders) ? Object.keys(decoders) : [];
```

Comment:
we don't need to decode `decoders` do we? Can just be

```ts
const decoderKeys = decoders ? Object.keys(decoders) : [];
```

Can we make `decoders` a required field? Then it would be even simpler:

```ts
const decoderKeys = Object.keys(decoders);
```